DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/mmhost/README.md

```diff
@@ -0,0 +1,354 @@
+
+![](assets/basic_avhost.svg)
+
+The compiler will infer Avalon memory-mapped host interfaces for your design when your kernel includes one or more pointer arguments. As with scalar kernel arguments, pointer arguments can be passed to your kernel via a `conduit` interface or your component's control/status register (CSR). By default, pointer arguments will be passed to your IP component through the CSR. For more details on kernel arguments, please see `<Temporary: https://github.com/intel-sandbox/wenkaixu.ip-auth-demos/tree/2024_0/mm-host>`. By default, the Intel® oneAPI DPC++/C++ Compiler will produce a kernel with a single Avalon memory-mapped host interface that will be shared amongst those pointers. 
+
+#### Example 1: A kernel with multiple pointer arguments
+```c++
+struct PointerIP{
+  int *x; 
+  int *y; 
+  int *z;
+  int size;
+
+  void operator()() const {
+    for (int i = 0; i < size; ++i) {
+      int add = x[i] + y[i];
+      z[i] = add;
+    }
+  }
+};
+```
+![](assets/visio-naive.svg)
+
+### Using `annotated_ptr` to configure Avalon memory-mapped host interfaces
+
+You can override the default behaviour of a pointer argument by declaring an `annotated_ptr` kernel argument instead.
+
+To use this feature, you must first make the necessary inclusions
+```c++
+#include <sycl/ext/oneapi/annotated_arg/annotated_ptr.hpp>
+```
+
+#### Example 2: A kernel with a single customized Avalon memory-mapped host interface
+```c++
+struct SingleMMIP {
+  annotated_ptr<int, decltype(properties{
+    register_map,
+    buffer_location<kBL1>,
+    awidth<32>, 
+    dwidth<32>, 
+    latency<0>, 
+    read_write_mode<read>,
+    maxburst<4>
+  })> my_pointer; 
+  void operator()() const {
+    ...
+    // Kernel code
+    my_pointer[x] = y;
+    ...
+  }
+}
+```
+
+The following table describes additional properties you can use to customize the interface. 
+
+| Parameter                 | Description
+|---                        |---
+| `register_map`            | Pass the pointer for this memory-mapped host interface through the IP component's control/status register
+| `conduit`                 | Pass the pointer for this memory-mapped host interface through a conduit interface 
+| `stable`                  | User guarantee that the pointer will not change between pipelined invocations of the kernel. The compiler uses this to furthur optimize the kernel.
```

Comment:
Should stable be moved to the next table?

Reply:
I was thinking that this property doesn't require a value to be set when specifying so it should be grouped with this table

Reply:
I like it more in the other table mainly because that would leave just `register_map` and `conduit` which are mutually exclusive here.

Reply:
Make note of the mutual exclusive nature of the first table, and move the stable argument to the other table